Automattic/wp-calypso
# Reader: fix site block UI

Fixes https://github.com/Automattic/wp-calypso/issues/19606.

@jancavan noticed that picking 'Block site' from the Reader post options menu no longer updated the UI with the 'blocked' post card:

<img width="832" alt="screen shot 2017-11-09 at 16 57 40" src="https://user-images.githubusercontent.com/17325/32618289-28ef79dc-c56f-11e7-8aad-7696dc0cbe42.png">

This PR addresses that glitch (caused by https://github.com/Automattic/wp-calypso/pull/19633#discussion_r149974275) and moves the handling of site block/unblock requests to the data layer.

### To test

Try blocking a feed backed by a WordPress.com site, like:

http://calypso.localhost:3000/read/feeds/32382

<img width="867" alt="screen shot 2017-11-09 at 17 01 35" src="https://user-images.githubusercontent.com/17325/32618529-e74aa3f2-c56f-11e7-8c18-e9e647203e92.png">

You should see blocked posts like this:

<img width="976" alt="screen shot 2017-11-09 at 16 57 37" src="https://user-images.githubusercontent.com/17325/32618515-dfef1868-c56f-11e7-972b-d4c0256a9c87.png">

Clicking 'undo' on any of the blocked posts should reverse the block.

### Known issues

If you block a site from the feed or site stream and then refresh the page, there'll be no sign it was ever blocked. This is a long-standing problem and is described in this issue: 

https://github.com/Automattic/wp-calypso/issues/19642

If you try and re-block the site, you'll get an error saying there was a problem with the site block.
